demo/Google.Storage.V1.Demo/Program.cs

```diff
@@ -13,175 +13,180 @@
 // limitations under the License.
 
 using Google.Apis.Download;
+using Google.Apis.Storage.v1;
+using Google.Apis.Storage.v1.Data;
 using Google.Apis.Upload;
-using Microsoft.Framework.Runtime.Common.CommandLine;
-using Newtonsoft.Json;
 using System;
+using System.Diagnostics;
 using System.IO;
-using System.Threading;
-using System.Threading.Tasks;
+using StorageObject = Google.Apis.Storage.v1.Data.Object;
+using System.Net.Http;
 
 namespace Google.Storage.V1.Demo
 {
-    /// <summary>
     /// Command line application to demonstrate features of the Google Cloud Storage client wrapper library.
-    /// </summary>
     public class Program
     {
-        /// <summary>
-        /// Entry point for the application.
-        /// </summary>
-        public int Main(string[] args)
+        public void Main(string[] args)
         {
-            var app = new CommandLineApplication
-            {
-                Name = "storage-demo",
-                Description = "Demonstrates the Google Cloud Storage client wrapper library",
-                FullName = "storage-demo - simple demonstration application for the Google Cloud Storage wrapper",                
-            };
+            var projectId = "YOUR-PROJECT-ID";
+
+            // create a file to test with
+            if (!File.Exists("demo-test.txt")) { File.WriteAllText("demo-test.txt", "testing 123..."); }
+
+            // create a connection to the GCS service
+            var client = StorageClient.Create();
+
+            // exercise Google Cloud Storage
+            ListBuckets(client, projectId);
+            Bucket bucket = CreateBucket(client, projectId, $"gcs-demo-bucket");
```

Comment:
Note that bucket IDs are *global*. (Surprising, I know.) So I'm not going to be able to do that now that you have.
For the demo app, I'd suggest including the project ID as a prefix. (Also there's no point in using an interpolated string here...)